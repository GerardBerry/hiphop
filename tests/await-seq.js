"use hopscript"

var rjs = require("../lib/reactive-js.js");

var sigA = new rjs.Signal("A");
var sigB = new rjs.Signal("B");
var sigO = new rjs.Signal("O");

var prg = <rjs.reactivemachine debug name="awaitseq">
  <rjs.inputsignal ref=${sigA}/>
  <rjs.inputsignal ref=${sigB}/>
  <rjs.outputsignal ref=${sigO}/>
  <rjs.sequence>
    <rjs.await signal_name="A"/>
    <rjs.await signal_name="B"/>
    <rjs.emit signal_name="O"/>
  </rjs.sequence>
</rjs.reactivemachine>;

exports.prg = prg;
