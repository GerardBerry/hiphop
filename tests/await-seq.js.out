---- reaction  GO:true RES:true INIT ----
---- SEL:false K0:false K1:true ----

---- reaction  GO:false RES:true ----
---- SEL:false K0:false K1:true ----

B;
---- reaction  GO:false RES:true ----
---- SEL:false K0:false K1:true ----

A;
---- reaction  GO:false RES:true ----
---- SEL:true K0:false K1:true ----

---- reaction  GO:false RES:true ----
---- SEL:false K0:false K1:true ----

---- reaction  GO:false RES:true ----
---- SEL:false K0:false K1:true ----

B;
---- reaction  GO:false RES:true ----
EMIT O
---- SEL:true K0:true K1:false ----

